qiskit/algorithms/time_evolvers/variational/solvers/ode/ode_function_factory.py

```diff
@@ -0,0 +1,82 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Abstract class for generating ODE functions."""
+from __future__ import annotations
+
+from abc import ABC
+from enum import Enum
+from typing import Callable
+
+import numpy as np
+
+from qiskit.circuit import Parameter
+
+from .abstract_ode_function import AbstractOdeFunction
+from .ode_function import OdeFunction
+
+from ..var_qte_linear_solver import VarQTELinearSolver
+
+
+class OdeFunctionType(Enum):
+    """Types of ODE functions for VatQTE algorithms."""
+
+    # more will be supported in the near future
+    STANDARD_ODE = "STANDARD_ODE"
+
+
+class OdeFunctionFactory(ABC):
+    """Factory for building ODE functions."""
+
+    def __init__(
+        self,
+        ode_function_type: OdeFunctionType = OdeFunctionType.STANDARD_ODE,
+        lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] | None = None,
+    ) -> None:
+        """
+        Args:
+            ode_function_type: An Enum that defines a type of an ODE function to be built. If
+                not provided, a default ``STANDARD_ODE`` is used.
+            lse_solver: Linear system of equations solver callable. It accepts ``A`` and ``b`` to
+                solve ``Ax=b`` and returns ``x``. If ``None``, the default ``np.linalg.lstsq``
+                solver is used.
```

Comment:
Th default here seems more to be a default of the VarQTELinearSolver. I am wondering if there is a better way, thinking that if the default were ever changed in that class this doc would need to be updated too which seems like a dependence we might perhaps want to avoid.

Reply:
True, I already removed some of these "fake defaults" in other parts of the code, but overlooked this one. The default part belongs to the `VarQTELinearSolver`.